src/cmds/qstat.c

```diff
@@ -302,47 +361,163 @@ states(char *string, char *q, char *r, char *h, char *w, char *t, char *e, int l
  * @param[in] n - attribute name
  * @param[in] r - resource name
  * @param[in] v - value for the attribute
+ * @param[in] one_line - one line per attribute
  *
  * @return Void
  *
  */
 void
-prt_attr(char *n, char *r, char *v)
-{
-	char *c;
+prt_attr(char *name, char *resource, char *value, int one_line) {
+	int first = 1;
+	int len = 0;
+	int start = 0;
+	int i = 0;
 	char *comma = ",";
-	int   first = 1;
-	int   l;
-	int   start;
-
-	start = strlen(n) + 7;	/* 4 spaces + ' = ' is 7 */
-	printf("    %s", n);
-	if (r) {
-		start += strlen(r) + 1;
-		printf(".%s", r);
-	}
-	printf(" = ");
-	c = strtok(v, comma);
-	while (c) {
-		if ((l = strlen(c)) + start < 78) {
-			printf("%s", c);
-			start += l;
-		} else {
-			if (! first) {
-				printf("\n\t");
-				start = 9;
+	char *buf = NULL;
+	char *keyvalpair = NULL;
+	char *key = NULL;
+	char *val = NULL;
+
+	if (value == NULL)
+		return;
+	switch (output_format) {
+	case FORMAT_JSON:
+		if (strcmp(name, ATTR_v) == 0) {
+			if (add_json_node(JSON_OBJECT, JSON_NULL, name, NULL) == NULL) {
+				fprintf(stderr, "qstat: out of memory\n");
+				exit(1);
+			}
+			keyvalpair = (char *) malloc(strlen(value) + 1);
+			if (keyvalpair == NULL) {
+				fprintf(stderr, "qstat: out of memory\n");
+				exit(1);
 			}
-			while (*c) {
-				putchar(*c++);
-				if (++start > 78) {
-					start = 8;
-					printf("\n\t");
+			while (*value) {
+				/* value is split based on comma and each key-value is stored in keyvalpair.
+				 * If the value contains an escaped comma, only the comma is copied to keyvalpair.  */
+				while (!(*value == *comma || *value == '\0')) {
+					if (*value == ESC_CHAR && *(value + 1) == *comma)
+						value++;
+					keyvalpair[i++] = *value++;
+				}
+				keyvalpair[i] = '\0';
+				key = strtok_r(keyvalpair, "=", &val);
```

Comment:
how about adding -
if (*value == '=') 
        key = value;
to the above while loop and doing away with the call to strtok_r() ?